Makefile

```diff
@@ -22,7 +22,7 @@ tar:
 
 exec:
 	mkdir -p usr/bin
-	gcc  $$(pkg-config --cflags python3) -Wno-unused-result -Wsign-compare -fmessage-length=0 -grecord-gcc-switches -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -g -DNDEBUG -fmessage-length=0 -grecord-gcc-switches -g -DOPENSSL_LOAD_CONF -fwrapv -fno-semantic-interposition src/reauth.c $$(pkg-config --libs python3) -o usr/bin/cloudguestregistryauth
+	gcc  $$(pkg-config --cflags python3-embed) -Wno-unused-result -Wsign-compare -fmessage-length=0 -grecord-gcc-switches -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -g -DNDEBUG -fmessage-length=0 -grecord-gcc-switches -g -DOPENSSL_LOAD_CONF -fwrapv -fno-semantic-interposition src/reauth.c $$(pkg-config --libs python3-embed) -o usr/bin/cloudguestregistryauth
```

Comment:
Fails on 15 SP2.
```
~> pkg-config --libs python3-embed
Package python3-embed was not found in the pkg-config search path.
Perhaps you should add the directory containing `python3-embed.pc'
to the PKG_CONFIG_PATH environment variable
No package 'python3-embed' found
```
cloud-regionsrv-client still has to support 15 SP2.

I agree with @dirkmueller the build questions should be addressed in a separate PR than the improvements for unit testing.

Reply:
No problem, I will split this.

Reply:
Hmm, concerning `python3-embed.pc` … it arrived only in https://github.com/python/cpython/pull/13500 (for Python 3.8; first released with tag `v3.8.0b1`). Now, the question is whether we really care about this functionality enough to make our Python non-standard. OTOH, it is just one file, which shouldn’t make much difference. @dirkmueller , opinions?

Reply:
It looks to me the only way to make this work in a more generic way is by passing in variable from the spec file, something like `CFLAGS_PYTHON` and `LIBS_PYTHON`. Then in the spec file when/if we switch the build to Python 3.11 we could change the value of the variables. Which is equally "cumbersome" as touching the makefile. The advantage of doing it in the spec file is that the change is front an center, while having the change in the Makefile means the developer has to remember that it's there without a hint. As such using variable in the Makefile and setting those through the spec file does have an advantage.